### PR TITLE
Dictionary cleanup

### PR DESCRIPTION
**Pull request by tbaranes**
Related to #52

### CODE REVIEW

File: Source/DictionaryExtension.swift [hunk 1]
```diff
@@ -41,51 +59,23 @@ public extension Dictionary {
             }
```
Comment (Sweeper777) [inline Source/DictionaryExtension.swift]: Can't we just do `self[key] = value` here?

File: Source/DictionaryExtension.swift [hunk 2]
```diff
@@ -41,51 +59,23 @@ public extension Dictionary {
...
 public extension Dictionary where Value: Equatable {
 
     public func difference(with dictionaries: [Key: Value]...) -> [Key: Value] {
         var result = self
-        for dictionary in dictionaries {
-            for (key, value) in dictionary {
+        dictionaries.forEach {
+            for (key, value) in $0 {
                 if result.has(key: key) && result[key] == value {
                     result.removeValue(forKey: key)
```
Comment (Sweeper777) [inline Source/DictionaryExtension.swift]: Why not do `result[key] = nil`?

File: README.md [hunk 3]
```diff
@@ -386,28 +376,38 @@ Easily get union of two dictionaries:
...
 ```
 
+Test each element against a closure:
+
+``` swift
+var dic = ["abc": "abc, "ab": "a", "b": "b"]
+let result = dic.testAll { key, _ in 
+    key.length < 3 
+}
+print(result) // ["ab": "a", "b": "b"]
```
Comment (Sweeper777) [inline README.md]: `result` should be `false` here, right? `testAll` returns a `Bool`.

**Review by Sweeper777 — CHANGES_REQUESTED** [review]

**Comment by tbaranes** [comment]
Fully agree with all your comments, all fixed 👍

**Review by Sweeper777 — APPROVED** [review]

**Pull request merged**